frontend/src/lib/components/Users/CreateUser.svelte

```diff
@@ -0,0 +1,104 @@
+  import { getSearchParamValues } from '$lib/util/query-params';
+  import { onMount } from 'svelte';
+  import { z } from 'zod';
+
+  export let autoLogin = true;
+  export let allowUsernames = false;
+  export let onSubmit: (() => void) | undefined = undefined;
+  export let submitButtonText = $t('register.button_register');
+  export let endpoint: 'register' | 'acceptInvitation' | 'createGuestUserByAdmin';
+
+  type RegisterPageQueryParams = {
+    name: string;
+    email: string;
+  };
+  let turnstileToken = '';
+  // $locale is the locale that our i18n is using for them (i.e. the best available option we have for them)
+  // getLanguageCodeFromNavigator() gives us the language/locale they probably actually want. Maybe we'll support it in the future.
+  const userLocale = getLanguageCodeFromNavigator() ?? $locale;
+  const formSchema = z.object({
+    name: z.string().trim().min(1, $t('register.name_missing')),
+    email: z.string().trim()
+      .min(1, $t('project_page.add_user.empty_user_field'))
+      .refine((value) => (allowUsernames && !value.includes('@')) || isEmail(value), { message: $t('form.invalid_email') }),
+    password: passwordFormRules($t),
+    score: z.number(),
+    locale: z.string().trim().min(2).default(userLocale),
+  });
+
+  let { form, errors, message, enhance, submitting } = lexSuperForm(formSchema, async () => {
+    const endpointHandler =
+        endpoint === 'acceptInvitation' ? acceptInvitation
+      : endpoint === 'register' ? register
+      : endpoint === 'createGuestUserByAdmin' ? createGuestUserByAdmin
+      : () => { throw new Error(`CreateUser doesn't know how to handle endpoint type ${endpoint}`) };
+    const { user, error } = await endpointHandler($form.password, $form.score, $form.name, $form.email, $form.locale, turnstileToken);
+    if (error) {
+      if (error.turnstile) {
+        $message = $t('turnstile.invalid');
+      }
+      if (error.accountExists) {
+        $errors.email = [$t('register.account_exists')];
+      }
+      if (error.invalidInput) {
+        $errors.email = [$t('register.invalid_input')];
+        // If we later allow username-only accounts to be created, add error message to $errors.username as well and change message to "specify email or username"
+      }
+      return;
+    }
+    if (user) {
+      if (onSubmit) onSubmit();
+      if (autoLogin) await goto('/home', { invalidateAll: true }); // invalidate so we get the user from the server
+      return;
+    }
+    throw new Error('Unknown error, no error from server, but also no user.');
+  });
+  onMount(() => { // query params not available during SSR
+    const urlValues = getSearchParamValues<RegisterPageQueryParams>();
+    form.update((form) => {
+      if (urlValues.name) form.name = urlValues.name;
+      if (urlValues.email) form.email = urlValues.email;
+      return form;
+    }, { taint: true });
+  });
+</script>
+
+<ProtectedForm {enhance} bind:turnstileToken>
```

Comment:
You're still using a turnstile protected form for creating guest users.
Maybe this would be a good use of `<svelte:component>` based on something like `export let protectedForm = false`;
(you can of course make `true` the default if you like. `false` as the default just makes things slightly "prettier" 😆)

Reply:
Ah right, I was meaning to remove the turnstile from the CreateUserModal. Thanks for catching that.

Reply:
Done in commit 0b9fd488.